samples/unit-tests/gantt/grid-axis/demo.js

```diff
@@ -2167,3 +2167,32 @@ QUnit.test('slotWidth', assert => {
         '#15742: Rightmost tick slotWidth should be much smaller than the other ticks'
     );
 });
+
+QUnit.test('Week dataTime format for useUTC: false, #16550.', assert => {
```

Comment:
Since this is a bug in `dateFormats`, it is probably not necessary to build a full fledged chart to test it. See how other [dateFormat tests are built](https://github.com/highcharts/highcharts/blob/master/samples/unit-tests/time/dateformat/demo.js), and add you case there instead. 

In addition, remember that when setting `useUTC` to false, the dates will be local. This may cause different results depending on in which timezone the CI is running our tests, and would typically be a difficult thing to debug. In the other tests we deal with this by either setting the time in a different way, or by making sure the test only runs in the CET timezone (which is convenient for Norway/Poland devs).

https://github.com/highcharts/highcharts/blob/4132352299f951424d424cc2221ef093501969a0/samples/unit-tests/time/dateformat/demo.js#L39

Reply:
PS: For a simplified, simplified demo, I think it boils down to something like this: https://jsfiddle.net/highcharts/df4x5308/